modules/RTC/RTC.js

```diff
@@ -143,6 +184,149 @@ export default class RTC extends Listenable {
+         *
+         * @returns {Promise}
+         */
+        function maybeRequestDesktopDevice() {
+            const devices = options.devices || [];
+            const isDesktopDeviceRequsted = devices.indexOf('desktop') !== -1;
+
+            return isDesktopDeviceRequsted
+                ? RTCUtils._newGetDesktopMedia(
+                    options.desktopSharingExtensionExternalInstallation,
+                    options.desktopSharingSources)
+                : Promise.resolve();
+        }
+
+        /**
+         * Instantiates a {@code JitsiLocalTrack} with the passed in desktop
+         * stream and pushes the model instance to the internal array
+         * localJitsiTracks to be returned later.
+         *
+         * @param {MediaStreamTrack} desktopStream - A track for a desktop
+         * capture.
+         * @returns {void}
+         */
+        function maybeCreateAndAddDesktopTrack(desktopStream) {
+            if (!desktopStream) {
+                return;
+            }
+
+            const { stream, sourceId, sourceType } = desktopStream;
+
+            localJitsiTracks.push(_newCreateLocalTrack({
+                stream,
+                sourceId,
+                sourceType,
+                track: stream.getVideoTracks()[0],
+                videoType: VideoType.DESKTOP
+            }));
+        }
+
+        /**
+         * Executes a request for audio and/or video, as specified in options.
+         * By default both audio and video will be captured if options.devices
+         * is not defined.
+         *
+         * @returns {Promise}
+         */
+        function maybeRequestCaptureDevices() {
+            const devices = options.devices || [ 'audio', 'video' ];
+            const requestedCaptureDevices = devices.filter(
+                device => device !== 'desktop');
+
+            if (!requestedCaptureDevices.length) {
+                return Promise.resolve();
+            }
+
+            const constraints = RTCUtils._newGetConstraints(
+                requestedCaptureDevices, options);
+
+            logger.info('Got media constraints: ', constraints);
+
+            return RTCUtils._newGetUserMediaWithConstraints(
+                requestedCaptureDevices, constraints);
+        }
+
+        /**
+         * Instantiates a {@code JitsiLocalTrack} with the passed in desktop
+         * stream and pushes the model instance to the internal array
+         * localJitsiTracks to be returned later.
+         *
+         * @param {MediaStreamTrack} avStream - A track for with audio and/or
+         * video track.
+         * @returns {void}
+         */
+        function maybeCreateAndAddAVTracks(avStream) {
+            if (!avStream) {
+                return;
+            }
+
+            if (avStream.getAudioTracks().length) {
+                const audioStream = new MediaStream();
+
+                avStream.getAudioTracks().forEach(
+                    t => audioStream.addTrack(t));
```

Comment:
can't addTrack get an array?

Reply:
Doesn't look like it, at least based on mdn docs for addTrack. However, the constructor might be able to take an array of tracks, which is maybe what you were remembering.